lib/marin/src/marin/rl/alerts.py

```diff
@@ -0,0 +1,352 @@
+        lesson_state: str,
+        graduation_performance: float | None = None,
+    ) -> AlertResult | None:
+        if lesson_state != "graduated":
+            return None
+
+        # Compute graduation performance from eval stats if not provided
+        if graduation_performance is None:
+            eval_rewards = stats.eval_stats.reward_history
+            if len(eval_rewards) == 0:
+                return None
+            # Use mean reward value at graduation time (before any regression)
+            # Strategy: Use mean of eval rewards excluding recent window (which might contain regression)
+            # This approximates graduation-time performance
+
+            # Use mean of eval rewards excluding the most recent window
+            # This assumes recent samples might contain regression, older samples represent graduation performance
+            recent_window = min(20, len(eval_rewards))
+            if len(eval_rewards) > recent_window:
+                # Use mean of older samples (before potential regression)
+                # Need at least a few samples for reliable baseline
+                graduation_rewards = eval_rewards[:-recent_window]
+                if len(graduation_rewards) > 0:
+                    graduation_performance = float(np.mean(graduation_rewards))
+                else:
+                    # Fallback: use all samples if split would be empty
+                    graduation_performance = float(np.mean(eval_rewards))
+            else:
+                # Not enough samples - use mean of all (fallback)
+                # With < 20 samples, can't reliably detect regression anyway
+                graduation_performance = float(np.mean(eval_rewards))
+
+        # For graduated lessons, compare recent mean eval reward (not all-time mean)
+        # Use recent window to detect current regression
+        if len(stats.eval_stats.reward_history) > 0:
+            eval_rewards = stats.eval_stats.reward_history
+            # Use recent eval rewards (last 20 samples) for current performance
+            recent_window = min(20, len(eval_rewards))
+            current_perf = float(np.mean(eval_rewards[-recent_window:]))
+        else:
+            # Fallback to training stats if no eval data
+            if len(stats.training_stats.reward_history) > 0:
+                train_rewards = stats.training_stats.reward_history
+                recent_window = min(20, len(train_rewards))
+                current_perf = float(np.mean(train_rewards[-recent_window:]))
+            else:
+                return None
+
+        if current_perf < graduation_performance * self.regression_threshold:
+            health_status = (
+                HealthStatus.CRITICAL
+                if current_perf < graduation_performance * self.critical_threshold
+                else HealthStatus.WARNING
+            )
+
+            # Calculate performance ratio safely (avoid division by zero)
+            if graduation_performance > 1e-6:
```

Comment:
We should disable the check up at the top if graduation_performance <= 1e-6, since there's no point monitoring the lesson if it's garbage already?